### PR TITLE
Add click-compatible jobs command

### PR DESCRIPTION
**Pull request by hankehly**
Related: #22708

## Summary

This PR adds click-compatible `jobs check` command.

Todo:
- [x] add click command
- [x] update unit tests

## Screenshots (before/after)

Screen Shot 2022-06-15 at 8 55 35

### CODE REVIEW

File: airflow/cli/commands/jobs.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+from airflow.utils.state import State
+
+
+@airflow_cmd.group("jobs")
+def jobs():
+    """Manage jobs"""
+
+
+@jobs.command("check")
+@click.option(
```
Comment (hankehly) [inline airflow/cli/commands/jobs.py:34]: Changes to this file:
- added `jobs()` command group
- added `@click.option` decorators to `check` function
- updated `check` function signature to include all arguments

File: tests/cli/commands/test_jobs.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        with create_session() as session:
+            self.scheduler_job = SchedulerJob()
+            self.scheduler_job.state = State.RUNNING
+            self.scheduler_job.hostname = 'HOSTNAME'
+            session.add(self.scheduler_job)
+            session.commit()
+            self.scheduler_job.heartbeat()
+
+        runner = CliRunner()
+        result = runner.invoke(jobs.check, ["--job-type", "SchedulerJob", "--hostname", "HOSTNAME"])
```
Comment (hankehly) [inline tests/cli/commands/test_jobs.py:61]: In this file, I only changed how `jobs.check` is invoked and how we check the output content / exception type.

**Comment by hankehly** [comment]
@blag Thanks for all the reviews. Please review this one too at your earliest convenience.

**Comment by blag** [comment]
The original help text had some examples in the ["epilog"](https://github.com/apache/airflow/blob/fe2ef0fcc75fa7be9e1ec1113af260a3a3c9e7ae/airflow/cli/cli_parser.py#L1761), and it would be nice to keep those in the new help text, even if they'll get rendered above the options.

**Comment by hankehly** [comment]
Thanks, I didn't see that. I'll add it in.

**Comment by hankehly** [comment]
@blag I added the epilog text to the `check` docstring. You can see the output in the updated screenshot.

**Review by blag — APPROVED** [review]

**Pull request merged**